controllers/reconcilers/prometheus_installation/prometheus_installation_reconciler.go

```diff
@@ -142,42 +159,56 @@ func (r *Reconciler) reconcileCatalogSource(ctx context.Context, cr *v1.Observab
 			Name:      "prometheus-catalogsource",
 			Namespace: cr.GetPrometheusOperatorNamespace(),
 		},
+		Spec: v1alpha1.CatalogSourceSpec{
+			Image: "quay.io/integreatly/custom-prometheus-index:1.0.0",
+		},
 	}
 	selector := client.ObjectKey{
 		Namespace: source.Namespace,
 		Name:      source.Name,
 	}
 
-	//look for catalogSource for old Prometheus Operator index. If found migrate to community image
+	// look for catalogSource for Prometheus Operator index. If not found check for unwanted Operator and remove
 	err := r.client.Get(ctx, selector, source)
-	if err != nil && !errors.IsNotFound(err) {
-		return v1.ResultFailed, err
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return v1.ResultFailed, err
+		}
+		err = r.removePrometheusOperatorResources(ctx, cr)
+		if err != nil {
+			return v1.ResultFailed, err
+		}
 	}
-	if err == nil {
-		if source.Spec.Image == "quay.io/integreatly/custom-prometheus-index:1.0.0" {
-			err := r.removePrometheusOperatorIndexResources(ctx, source, cr)
-			if err != nil {
-				return v1.ResultFailed, err
-			}
-			return v1.ResultSuccess, nil
+
+	source = model.GetPrometheusCatalogSource(cr)
```

Comment:
Seems like this returns the same as line 157 and 161? 🤔

Reply:
Yeah good point they are returning same once `source.Spec.Image` is removed. Thanks